### PR TITLE
change Basic.NaN to S.NaN

### LINKED ISSUE #4841: how to implement functions for sympy
```
From doc/src/guide.txt:

[...]
if isinstance(arg, Basic.NaN):
    return S.NaN
if isinstance(arg, Basic.Zero):
    return S.Zero
[...]

I have several questions:

Why using isinstance()?
Why mixing S.Zero and Basic.Zero?
Why not using

if arg is S.NaN: ...

?
```

Original issue for #4841: http://code.google.com/p/sympy/issues/detail?id=1742
Original author: https://code.google.com/u/Vinzent.Steinberg@gmail.com/
Original owner: https://code.google.com/u/Vinzent.Steinberg@gmail.com/

### PR DESCRIPTION
**Pull request by namannimmo10**
#### References to other Issues or PRs

Fixes #4841

#### Brief description of what is fixed or changed
type object 'Basic' has no attribute 'NaN'.
type object 'Basic' has no attribute 'Zero'

#### Release Notes

NO ENTRY

### CODE REVIEW

File: doc/src/guide.rst [hunk 1]
```diff
@@ -285,9 +285,9 @@ How to create a new function with one variable::
 
         @classmethod
         def eval(cls, arg):
-            if isinstance(arg, Basic.NaN):
+            if isinstance(arg, S.NaN):
```
Comment (ShubhamKJha) [inline doc/src/guide.rst]:
```suggestion
            if arg is S.NaN:
```

File: doc/src/guide.rst [hunk 2]
```diff
@@ -285,9 +285,9 @@ How to create a new function with one variable::
...
                 return S.NaN
-            if isinstance(arg, Basic.Zero):
+            if isinstance(arg, S.Zero):
```
Comment (ShubhamKJha) [inline doc/src/guide.rst]: similarly here

**Review by ShubhamKJha — CHANGES_REQUESTED** [review]

File: doc/src/guide.rst (hunk 1 above)
Comment (namannimmo10) [reply doc/src/guide.rst]: sure. I'll update

**Comment by sylee957** [comment]
Can those tutorial be doctestable?

**Comment by namannimmo10** [comment]
> Can those tutorial be doctestable?

Explain doctestable?? Maybe extracting the code from docs and testing it??

**Comment by sylee957** [comment]
If you change some examples like
```python3
>>> class sign(Function):
...     nargs = 1
...
```
and then it's going to be picked up by our test runner to actually run the code and verify that there is no syntax error or such.

One other advantage by adding some `>>> ` is that you can actually run the piece with the sympy live shell.

If you see some code examples below that can be clickable https://docs.sympy.org/latest/guide.html, you'll know what I mean.

**Review by ShubhamKJha — APPROVED** [review]

**Comment by namannimmo10** [comment]
@sylee957 thanks for the tips!

**Comment by ShubhamKJha** [comment]
You can run `doctest` your changes locally by using
```
python bin\doctest sympy\doc
```
> Can those tutorial be doctestable?

The code can be easily made to be doctested, but the error (here `Basic.NaN` doesn't exist) is not caught because the class is not initialized. Don't know how many such cases would be there.

**Comment by oscarbenjamin** [comment]
> because the class is not initialized

The example would be better if it was complete with initialisation and usage of the class.

**Comment by oscarbenjamin** [comment]
This is a good contribution. Thank you @namannimmo10. I'm merging this now.

**Comment by namannimmo10** [comment]
> This is a good contribution. Thank you @namannimmo10. I'm merging this now.

Thanks! Feels good ;)

**Pull request merged**